mitodl/open-discussions
# Use default profile image if the user does not exist in open-discussions

#### What are the relevant tickets?
Part of #184 

#### What's this PR do?
Works around exception when user does not exist in open-discussions:
 - in a list of posts (frontpage or channel view) the ones missing a user should be removed from the list as if they were deleted
 - in a list of comments the comments missing a user should show up as '[deleted]'
 - if a post or comment has a missing user it should return a 404 in the detail view

Note that the text or url will still show up for comments and posts even if the author is `[deleted]`.

#### How should this be manually tested?
Make a post in open-discussions and delete that user from the database. View a post by that user. You should see the default avatar image.

Make a comment in open-discussions. Delete the open-discussions user who made the comment. View the comment in `/api/v0/posts/<post_id>/comments/` and it should show up with `[deleted]` as the author name.

One more case: make a comment in open-discussions and make a child comment of that comment. In reddit delete the parent comment. View the comments in the REST API and you should still see the child comment, with the parent comment showing up as `[deleted]` and text `[deleted]`
